### PR TITLE
feature(meeting-container): adds meeting container 

### PR DESCRIPTION
**Pull request by breardon2011**
Adds in MeetingContainer for space back and non space backed meetings. 

# Description
A new component was added to handle both space backed and non spacebacked meetings. It extends ButtonSimple and Card. It was implemented in accordance with Figma. 

Card was modified by adding isStatic and isDisabled props to handle the disabled state coming from the MeetingContainer component 

Tests were added to handle new states, snapshot and check for extension of ButtonSimple. 

# Links
[Figma](https://www.figma.com/file/NaNrfXjygZtRgMfHAFHjsp/branch/3AnFAJsgv2FnOkeQejbVUU/Components---Windows?node-id=27314%3A50060)

[Space backed Jira](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-313910)

[Not Space backed Jira](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-313912)

### CODE REVIEW

File: src/components/Card/Card.unit.test.tsx [hunk 1]
```diff
@@ -248,5 +268,40 @@ describe('<Card />', () => {
...
+    });
+
+    it('should extend ButtonSimple', async () => {
+      expect.assertions(1);
+
+      render(<Card data-testid={testid} />);
+
+      const component = await screen.findByTestId(testid);
+
+      expect(component.classList.contains(BUTTON_SIMPLE_CONSTANTS.STYLE.wrapper)).toBe(true);
```
Comment (InteractiveTimmy) [inline src/components/Card/Card.unit.test.tsx:304]: Good way to validate this!

File: src/components/MeetingContainer/MeetingContainer.constants.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  left: `${CLASS_PREFIX}-left`,
+  right: `${CLASS_PREFIX}-right`,
+  padding: `${CLASS_PREFIX}-padding`,
+  title: `${CLASS_PREFIX}-title`,
+  spaceLink: `${CLASS_PREFIX}-spaceLink`,
+  date: `${CLASS_PREFIX}-date`,
+  convo: `${CLASS_PREFIX}-convo`,
+  avatar: `${CLASS_PREFIX}-avatar`,
+  now: `${CLASS_PREFIX}-now`,
+  container: `${CLASS_PREFIX}-container`,
```
Comment (InteractiveTimmy) [inline src/components/MeetingContainer/MeetingContainer.constants.ts:37]: This area may need to be cleaned up, I don't think all of these are used in the component.

File: src/components/MeetingContainer/MeetingContainer.stories.docs.mdx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+The `<MeetingContainer />` component.
+
+This component displays meeting details and action buttons supplied by the consumer
```
Comment (InteractiveTimmy) [inline src/components/MeetingContainer/MeetingContainer.stories.docs.mdx:3]: Great mdx writeup, contains more than enough information to understand the component! good work!

File: src/components/MeetingContainer/MeetingContainer.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+const spaceLink = <ButtonHyperlink>Example Link</ButtonHyperlink>;
+
+const avatar = <Avatar>BR</Avatar>;
+
+const continueConversation = <ButtonPill size={20}>Continue the conversation</ButtonPill>;
+
+Example.argTypes = { ...argTypes };
+
+// TODO: Provide default arguments for this story here. These populate into the argument table for this component.
```
Comment (InteractiveTimmy) [inline src/components/MeetingContainer/MeetingContainer.stories.tsx:65]: We can remove these comments, as they are not necessary after the component is completed. Any comments including `TODO` or `NOTE`, unless explicitly written by yourself to help better-define code, should be removed.

File: src/components/MeetingContainer/MeetingContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import ButtonGroup from '../ButtonGroup';
+import { action } from '@storybook/addon-actions';
+
+/**
+ * The MeetingContainer component.
+ *
+ * Will render avatar/tags/action buttons if provided.
+ *
+ * Depends on consumer for schedule info strings/statusColor
+ *
```
Comment (InteractiveTimmy) [inline src/components/MeetingContainer/MeetingContainer.tsx:18]: You can remove this line.

File: src/components/MeetingContainer/MeetingContainer.unit.test.tsx [hunk 6]
```diff
@@ -0,0 +1,477 @@
...
+      .at(0)
+      .getDOMNode();
+
+    expect(element.innerHTML).toBe(meetingTitle);
+  });
+  // it should have as spaceLink when provided
+  it('should have spaceLink when provided', () => {
+    const element = mount(<MeetingContainer spaceLink={spaceLink} meetingTitle={meetingTitle} />)
+      .find(ButtonHyperlink)
+      .exists();
```
Comment (InteractiveTimmy) [inline src/components/MeetingContainer/MeetingContainer.unit.test.tsx:458]: Great overall test coverage!

**Review by InteractiveTimmy — CHANGES_REQUESTED** [review]
Looks great! A few blocking changes around cleaning up comments and constants.

**Comment by breardon2011** [comment]
I also just realized I forgot to put exports in index files initially

File: src/components/MeetingContainer/MeetingContainer.stories.args.ts [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    table: {
+      type: {
+        summary: 'string',
+      },
+      defaultValue: {
+        summary: 'undefined',
+      },
+    },
+  },
+  titleType: {
```
Comment (InteractiveTimmy) [inline src/components/MeetingContainer/MeetingContainer.stories.args.ts:19]: Looks like you're missing `meetingTitle` prop from the argTypes?

**Review by InteractiveTimmy — CHANGES_REQUESTED** [review]
Looks like you're missing one prop from the argTypes? Correct me if I'm mistaken.

**Review by InteractiveTimmy — APPROVED** [review]
LGTM!

**Pull request merged**